server/lib/src/idm/group.rs

```diff
@@ -70,30 +71,34 @@ macro_rules! try_from_account_e {
                 vec![]
             }
         };
-        groups.push(upg);
-        Ok(groups)
+        if let Some(mut non_empty_groups) = NonEmpty::collect(groups.into_iter()) {
+            non_empty_groups.push(upg);
+            Ok(non_empty_groups)
+        } else {
+            Ok(nonempty!(upg))
+        }
```

Comment:
This seems a bit convoluted, since we always want `upg` to be a member, you could do:

```rust
non_empty_groups = NonEmpty{ head: upg, tail: groups }
```

Reply:
heh, closed while I was doing this. 🤷🏻